iron-overlay-behavior.html

```diff
@@ -187,6 +187,16 @@
         window.cancelAnimationFrame(this.__raf);
         this.__raf = null;
       }
+      if (this.__isAnimating) {
+        this.__isAnimating = false;
+        // We got detached while animating, ensure styles are ok
+        // in case we get ever attached again.
+        // NOTE: it's too late to fire iron-overlay-closed as we're already detached.
```

Comment:
Yes, we could fire the event (e.g. use the host as the node from which to fire the event).
But I don't think it would be semantically correct to notify "the overlay was closed" when its animation is not done yet...similar to the scenario where an animating element that gets detached before finishing the animation wouldn't fire a "transitionend" event. WDYT?